engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/URLValidatorTest.java

```diff
@@ -290,10 +290,91 @@ private void runUrlContainerValidation(Validator validator, URLContainer contain
+		assertTrue( validator.isValid( "http://j.mp", null ) );
+		assertTrue( validator.isValid( "ftp://foo.bar/baz", null ) );
+		assertTrue( validator.isValid( "http://foo.bar/?q=Test%20URL-encoded%20stuff", null ) );
+		assertTrue( validator.isValid( "http://مثال.إختبار", null ) );
+		assertTrue( validator.isValid( "http://例子.测试", null ) );
+		assertTrue( validator.isValid( "http://उदाहरण.परीक्षा", null ) );
+		assertTrue( validator.isValid( "http://-.~_!$&'()*+,;=:%40:80%2f::::::@example.com", null ) );
+		assertTrue( validator.isValid( "http://1337.net", null ) );
+		assertTrue( validator.isValid( "http://a.b-c.de", null ) );
+		assertTrue( validator.isValid( "http://223.255.255.254", null ) );
+
+		// invalid urls:
 		assertFalse( validator.isValid( "http", null ) );
 		assertFalse( validator.isValid( "ftp//abc.de", null ) );
-		assertTrue( validator.isValid( "ftp://abc.de", null ) );
+		assertFalse( validator.isValid( "//", null ) );
+		assertFalse( validator.isValid( "//a", null ) );
+		assertFalse( validator.isValid( "///", null ) );
+		assertFalse( validator.isValid( "///a", null ) );
+		assertFalse( validator.isValid( "foo.com", null ) );
+		assertFalse( validator.isValid( ":// should fail", null ) );
+//		assertFalse( validator.isValid( "rdar://1234", null ) );
+//		assertFalse( validator.isValid( "ftps://foo.bar/", null ) );
+//		assertFalse( validator.isValid( "h://test", null ) );
+
+		if ( validator instanceof RegexpURLValidator ) {
+//			assertFalse( validator.isValid( "http://", null ) );
+//			assertFalse( validator.isValid( "http://.", null ) );
+//			assertFalse( validator.isValid( "http://..", null ) );
+//			assertFalse( validator.isValid( "http://../", null ) );
+//			assertFalse( validator.isValid( "http://?", null ) );
+//			assertFalse( validator.isValid( "http://??", null ) );
+//			assertFalse( validator.isValid( "http://??/", null ) );
+//			assertFalse( validator.isValid( "http://#", null ) );
+//			assertFalse( validator.isValid( "http://##", null ) );
+//			assertFalse( validator.isValid( "http://##/", null ) );
+			assertFalse( validator.isValid( "http://foo.bar?q=Spaces should be encoded", null ) );
+//			assertFalse( validator.isValid( "http:///a", null ) );
+			assertFalse( validator.isValid( "http:// shouldfail.com", null ) );
+			assertFalse( validator.isValid( "http://foo.bar/foo(bar)baz quux", null ) );
+//			assertFalse( validator.isValid( "http://-error-.invalid/", null ) );
+//			assertFalse( validator.isValid( "http://a.b--c.de/", null ) );
+//			assertFalse( validator.isValid( "http://-a.b.co", null ) );
+//			assertFalse( validator.isValid( "http://a.b-.co", null ) );
+//			assertFalse( validator.isValid( "http://0.0.0.0", null ) );
+//			assertFalse( validator.isValid( "http://10.1.1.0", null ) );
+//			assertFalse( validator.isValid( "http://10.1.1.255", null ) );
+//			assertFalse( validator.isValid( "http://224.1.1.1", null ) );
```

Comment:
@gunnarmorling the link from which the URLs were extracted was used to have an URL validator for link shortener. Thus they have very special requirements. That's why we should be careful with the examples given.